var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -110,9 +110,9 @@ class Protobuf(CMakePackage):
 
     # fix build on Centos 8, see also https://github.com/protocolbuffers/protobuf/issues/5144
     patch(
-        "https://github.com/protocolbuffers/protobuf/commit/3039f932aaf212bcf2f14a3f2fd00dbfb881e46b.patch?full_index=1",
+        "https://github.com/protocolbuffers/protobuf/commit/462964ed322503af52638d54c00a0a67d7133349.patch?full_index=1",
```

Comment:
Maybe I don't understand. These are not the URLs affected by the redirect dropping the full_index; only URLs with a pull request reference do. The original URL has full indices and the sha256 is still the same here. So, I don't see what this fixes since it's not this URL that can have been the issue to begin with.

Reply:
See https://github.com/spack/spack/pull/44210 for original discussion, by the way.

Reply:
:facepalm: I copied the url from the top level comment.

I think @tombradford ran into this issue on a version of Spack older than 9bcc43c4c158639fa6cb575c6106595a34682081 and submitted a PR not realizing it was already fixed.

In any case, the current change is imo good, because https://github.com/protocolbuffers/protobuf/commit/462964ed322503af52638d54c00a0a67d7133349 is not part of any branch, whereas https://github.com/protocolbuffers/protobuf/commit/462964ed322503af52638d54c00a0a67d7133349 is.

Reply:
Many thanks! No, I hadn't realised this was already fixed (sorry!), as we have forked the 0.22 release on our cluster. Next time, I'll try a build on develop before raising an issue/PR.

Reply:
I completely agree that commits in the target branch are better. Thanks!